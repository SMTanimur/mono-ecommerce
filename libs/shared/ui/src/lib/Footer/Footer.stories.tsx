import { Story, Meta } from '@storybook/react';
import { Footer } from './Footer';

export default {
  component: Footer,
  title: 'Components/Footer',
} as Meta;

const Template: Story = (args) => <Footer {...args} />;

export const Primary = Template.bind({});
