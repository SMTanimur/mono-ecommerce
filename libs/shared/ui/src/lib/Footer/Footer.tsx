import React from 'react';

import clsx from 'clsx';

export const Footer: React.FC = () => {
    return (
        <footer className={clsx('mt-20 w-full', 'bg-gray-100')}>
            <div
                className={clsx(
                    'mx-auto max-w-screen-xl py-8 px-4',
                    'flex gap-8 items-center justify-between',
                )}
            >
                <ul
                    className={clsx(
                        'flex gap-4',
                        'text-xs not-italic',
                        'opacity-65',
                    )}
                >
                    <li>
                        <a
                            className="hover:underline"
                            rel="noreferrer noopener"
                            target="_blank"
                            href="https://github.com/artusm/mono-ecommerce"
                        >
                            Github
                        </a>
                    </li>
                    <li>
                        <a
                            className="hover:underline"
                            rel="author noreferrer noopener"
                            target="_blank"
                            href="https://github.com/artusm"
                        >
                            artusm
                        </a>
                    </li>
                </ul>
            </div>
        </footer>
    );
};
