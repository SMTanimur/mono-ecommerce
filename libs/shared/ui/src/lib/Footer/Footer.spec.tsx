import { render, RenderResult, screen } from '@testing-library/react';
import {create} from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Footer } from './Footer';

describe('Footer', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Footer />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('should have author link', () => {
    expect(screen.getByRole('link', { name: 'artusm' })).toBeTruthy();
  });
});
