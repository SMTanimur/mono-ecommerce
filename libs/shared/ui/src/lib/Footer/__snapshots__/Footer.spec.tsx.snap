// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render successfully 1`] = `
<footer
  className="mt-20 w-full bg-gray-100"
>
  <div
    className="mx-auto max-w-screen-xl py-8 px-4 flex gap-8 items-center justify-between"
  >
    <ul
      className="flex gap-4 text-xs not-italic opacity-65"
    >
      <li>
        <a
          className="hover:underline"
          href="https://github.com/artusm/mono-ecommerce"
          rel="noreferrer noopener"
          target="_blank"
        >
          Github
        </a>
      </li>
      <li>
        <a
          className="hover:underline"
          href="https://github.com/artusm"
          rel="author noreferrer noopener"
          target="_blank"
        >
          artusm
        </a>
      </li>
    </ul>
  </div>
</footer>
`;
